### DISCUSSION TITLE
Automatic "Reset" of DKIM?

### DISCUSSION THREAD

**Discussion by btalin** [postalserver/postal#2595] [Help with using Postal]
## Describe the bug

There is a strange bug. Sometimes Postal just creates a new "DKIM" 
e.g. for a domain it was 

postal-ogenqi._domainkey with a DKIM

and now suddenly I saw that the DKIM is not set right and Postal marked it with "X" in the domain overview and suddenly there is a new domain key with DKIM

postal-wgrb0p._domainkey

## To Reproduce

Unfortunately it just happens. We didnd't restart, we didn't change anything, it is running for weeks now like it is.

## Expected behaviour

DKIM should stay the same right?

**Comment by willpower232** [comment]
I don't believe it is possible for this to happen randomly. I have seen people get confused when adding the same domain to multiple servers within Postal so it gets a different DKIM record for each use.

**Reply by yneophytou** [reply]
Facing the same issue as well since last Thursday  24th August 2023. DKIM record changes multiple time in a day without any notice etc.

**Reply by btalin** [reply]
I checked everything now. The domain is only with 1 server connected and none of the servers has another domain or second domain mentioned. So 1 Server 1 Domain and still it happens randomly. Now was good for 3 days but had it several times and needed to update the DKIM about 4 times till now even I changed nothing on the server, no restart, nothing.

**Comment by jmdunsing** [comment]
I'll add that I also noticed a server that, after some time, the DKIM selector decided to change to "postal". See below.

![image](https://github.com/postalserver/postal/assets/113212353/17e3d20b-2608-45f8-aeac-fd85e72cb56f)

I noticed that after running the DNS check on the domain, it fixed the the selector and emails are now going through. Again, I'm sure I've done this check in the past, but postal had no record of it ever being done. I personally this is a bug, but at least running the check seems to be a workaround.

**Reply by krystalcode** [reply]
This happened to me as well on a new setup. The 2nd time I noticed it just after adding the first route on the domain, but it could be coincidence. The domain that it happened on is `example.com` where Postal is installed at `postal.example.com`. When I added the route, in the dropdown for the route name I see the domain listed twice; once as a server domain and once as an organization domain. Is that setup wrong? I did not add the domain somewhere specifically as the server domain, should the parent domain of the postal installation not be used as an organization domain?

**Reply by krystalcode** [reply]
@willpower232 Could you explain the difference between server domain and organization domain? Why is it displayed two times in the dropdown when creating/editing a route? This issue happens to me only for that domain - DKIM changes all the time. It does not happen for other domains. I do not recall creating the domain two times.
